### PR TITLE
[COM-297] Modal component

### PR DESCRIPTION
**Pull request by dawinValenzuela**
## Please delete checklist items and sections that are not relevant

**Checklist**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)
https://impulsumvc.atlassian.net/browse/COM-297

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

**Other information**:

**Storybook Link** (Link to a local storybook url to review new functionality)
http://localhost:8080/?path=/story/components-primitives-modal--all

### CODE REVIEW

**Review by tkanzakic — CHANGES_REQUESTED** [review]
storybook is not building because of a type error

File: packages/components/src/components/primitives/Modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    </ChakraModalFooter>
+  );
+};
+
+Modal.Overlay = ModalOverlay;
+Modal.Content = ModalContent;
+Modal.Header = ModalHeader;
+Modal.Footer = ModalFooter;
+Modal.Body = ModalBody;
+Modal.CloseButton = ModalCloseButton;
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Modal/Modal.tsx:111]: I tend to remember we talked about not doing this but accepting props

File: packages/components/src/components/primitives/Modal/Modal.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        </Modal.Content>
+      </Modal>
+    </>
+  );
+};
+
+export const Playground = Template.bind({});
+
+Playground.args = {
+  size: ModalSizes[0],
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Modal/Modal.stories.tsx:147]: please add more config options

**Review by tkanzakic — CHANGES_REQUESTED** [review]
Some errors found while testing

Screenshot 2022-02-01 at 14 59 04
Screenshot 2022-02-01 at 14 59 47

**Review by TheEnmanuel23 — COMMENTED** [review]
**UNOFFICIAL REVIEW:**
After clicking on 20rem and right aligned button:
image
It seems to have left alignment instead of right alignment

**Review by guilean — APPROVED** [review]

**Review by tkanzakic — APPROVED** [review]

**Pull request merged**